### DISCUSSION TITLE
SK6812 wiring question

### DISCUSSION THREAD

**Discussion by jingato** [awawa-dev/HyperHDR#717] [General]
Hey,

So I just received my SK6812 and am a little confused about the wiring when comparing to the examples in the guide. For reference, I am using the Pimoroni Plasma 2040 as the driver so no need for the additional level shifter. However, the examples in the guide all show how to wire it directly to the led strip 3 connection points. However, my strip has a 3 pin connector as well as 2 additional wires coming out. I am pretty sure the two additional wires are the power and the ground, but then what do I do with the other 3 pin since the data is only a single wire? So I just ignore the other wires in the 3 pin connector? Or is the other wires in the 3 pin also power? If it is, does that mean I can only use the 3 pin and ignore the other 2 or do I need power to both?

The other question is regarding the common ground. I know I need to also connect the ground from the power supply to the driver, but does it come directly rom the power supply do I put the power and ground from the power supply only to the led strip and then set one of those other wires ( assuming it is connected to the ground ) coming out of the led strip to the driver?

Thanks!

**Comment by jingato** [comment]
I found the answer on the Amazon page's QA section. Basically the 3 pin is the power, ground, and data. The extra 2 wires are also power, but only needed for supplying extra power when daisy chaining strips. I plan on using the existing 3 pin input which has the power, ground, and data and then also sending power and ground to the other end of the strip to the two extra wires there. I will also send the ground back to the driver right from the power supply.

**Comment by awawa-dev** [comment]
Hi

One of input on the LED strip is a ground (usually marked as GND) and that line must have a connection also to ground of the LED controller: it doesn't necessarily have to come directly from the LEDs, but it's the same circuit. Should be as short as possible, similar to data line.

I do not recommend using the plugs that come with LEDs, no one can guarantee you how they are connected underneath, whether there is a short circuit between the cables where they are soldered to the LED strip, and certainly that a given color means a given signal line: almost all of them are cheap products from China => no rules. Moreover, their cables are rather thin, they don't have much in common with copper, so they can get hot if you try to power a large setup using only one input (that's probably why they add a second connector to get around their low quality). But it's your decision.

**Reply by jingato** [reply]
Ah, ok. Good to know. I'll just rip those off and solder my own in that case. Should be simple. I even got a bunch of 3 pin connectors so I can just add my own. I need to do it for each corner section anyway.

I did see you mentioning the data cable being as short at possible in your guide. My Pi will be about 1 meter from the connection of the LED strip. Is that too long? My other option would be to just put stick my driver to the back of my TV using double sides tape (I had a case 3D printed for it) and run my USB up there, which would allow me to make my both the data and the ground to the driver very, very short. Do you think that is worth doing?

**Reply by awawa-dev** [reply]
>  My Pi will be about 1 meter from the connection of the LED strip. Is that too long?

Pi (USB to Pimoroni) or Pimoroni Plasma 2040 (data signal and ground to the LED)? The USB cable can be very long, depending on its quality. There are also USB amplifiers and USB hubs. Data signal & ground? who knows, it may depend on the environment and EMI interference where it is placed. I used a maximum of ~70cm (with a level shifter, of course).

**Reply by jingato** [reply]
Yeah, I was going to go with the data line being 1 meter long and a 4 cm USB cable. I think I am going to scrap that idea and go with the longer USB and a really short data cable though. Thanks again!
